lib/irb/command/cd.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+module IRB
+  module Command
+    class CD < Base
+      category "Workspace"
+      description "Move into the given object or leave the current context."
+
+      help_message(<<~HELP)
+        Usage: cd ([target]|..|-)
+
+
+        When given:
+        - an object, cd will move into that object, making it the current context.
+        - "..", cd will leave the current context, moving back to the previous context.
+        - "-", cd will toggle between the last 2 contexts. This does not work with other workspace commands.
+        - no arguments, cd will move back to the top-level (main object) context.
+
+        Examples:
+
+          cd Foo
+          cd Foo.new
+          cd @ivar
+          cd ..
+          cd -
+          cd
+      HELP
+
+      def execute(arg)
+        previous_workspace = irb_context.workspace
+
+        case arg.strip
+        when ".."
+          irb_context.pop_workspace
+        when "-"
+          irb_context.replace_workspace(irb_context.previous_workspace)
```

Comment:
To support this, we need a way to remember popped workspaces too. And because commands are not memoized, we can't use ivars here. So I needed to add a new `previous_workspace` attribute to `Context` just for this feature. Additionally, it will not work with other commands like `pushws` or `chws` without adding further complexities.

While I understand its use cases, I personally don't use it. So either keeping or dropping it is fine for me. Thoughts? @ruby/irb-reline

Reply:
I think it's worth adding, or maybe we can add it later. Not particular about keeping or dropping it at the first step.

Since `cd` has a stack that represents directory-like model, I think `cd -` should restore the whole stack, not just the current workspace.
```
cd 1  # 1
cd 2  # 1/2
cd 3  # 1/2/3
cd .. # 1/2
cd -  # 1/2/3
cd    # main
cd -  # 1/2/3
cd .. # 1/2
```

And we need to consider `irb_context.previous_workspace==nil` case.
```
$ irb
irb(main):001> cd -
```

Reply:
That's a good point. I'll update my implementation 👍